### PR TITLE
Check that pairings work properly when the G1 point is at infinity.

### PR DESCRIPTION
**Pull request by ebfull**

### CODE REVIEW

File: src/gtest/test_proofs.cpp [hunk 1]
```diff
@@ -22,6 +22,51 @@ typedef libsnark::default_r1cs_ppzksnark_pp::Fqe_type curve_Fq2;
 #include "version.h"
 #include "utilstrencodings.h"
 
+TEST(proofs, g1_pairing_at_infinity)
```
Comment (str4d) [inline src/gtest/test_proofs.cpp]: Use `CamelCase` instead of `snake_case`.

**Comment by nathan-at-least** [comment]
@zkbot try

**Comment by nathan-at-least** [comment]
@zkbot try

**Comment by nathan-at-least** [comment]
@zkbot retry

**Comment by nathan-at-least** [comment]
@zkbot retry

**Comment by nathan-at-least** [comment]
@zkbot r+ p=-17

**Comment by str4d** [comment]
@zkbot p=1

**Pull request merged**